File: Tipos de Dados/Ordinais/Ordinais.pas
//Pascal
//Em Pascal, tanto enumerações quanto subfaixas são suportadas.
//Pascal
//Em Pascal, tanto enumerações quanto subfaixas são suportadas.
program Ordinais;

type
    DiaDaSemana = (Segunda, Terca, Quarta, Quinta, Sexta, Sabado, Domingo);
    Nota = 1..10;  { Subfaixa de 1 a 10 }

var
    hoje: DiaDaSemana;
    minhaNota: Nota;

begin
    hoje := Terca;
    minhaNota := 8;

    // Exibindo valor ordinal
    writeln('Hoje é: ', Ord(hoje));

    // Usando a posição ordinal
    writeln('Posição ordinal: ', Ord(hoje));

    if hoje = Terca then
        writeln('Hoje é terça-feira.');

	// Verificar se a nota está dentro dos limites
	if (minhaNota >= 1) and (minhaNota <= 10) then
		writeln('A nota está dentro da subfaixa permitida.')
	else
		writeln('A nota está fora da subfaixa permitida.');
	
end.

File: Tipos de Dados/Ordinais/ordinais.php
<?php
//PHP
//PHP 8.1 introduziu o suporte a enumerações.
enum DiaDaSemana: int {
    case Segunda = 0;
    case Terca = 1;
    case Quarta = 2;
    case Quinta = 3;
    case Sexta = 4;
    case Sabado = 5;
    case Domingo = 6;
}

$hoje = DiaDaSemana::Terca;

// Exibindo valor ordinal
echo "Hoje é: " . $hoje->name . "\n";

// Usando a posição ordinal
echo "Posição ordinal: " . $hoje->value . "\n";

if ($hoje === DiaDaSemana::Terca) {
    echo "Hoje é terça-feira.\n";
}
?>
